app/components/sidebar/TenantSelection.tsx

```diff
@@ -13,60 +13,72 @@ import SideBarDropdownItem from './SideBarDropdownItem';
 
 const t = StageUtils.getT('users');
 
+const StyledLoaderWrapper = styled.div`
+    height: 80px;
+`;
+
 const TenantSelection: FunctionComponent = () => {
     const [search, setSearch, clearSearch] = useResettableState('');
     const manager = useSelector((state: ReduxState) => state.manager || {});
+    const { tenants } = manager;
     const dispatch = useDispatch();
 
-    function onTenantsRefresh() {
+    const isLoadingTenants = !tenants.items || tenants.isFetching;
+
+    function refreshTenants() {
         dispatch(getTenants());
     }
 
-    useEffect(() => {
-        const eventName = 'menu.tenants:refresh';
-        EventBus.on(eventName, onTenantsRefresh);
-        return () => EventBus.off(eventName, onTenantsRefresh);
-    }, [onTenantsRefresh]);
-
     function onTenantSelected(tenant: { name: string }) {
         dispatch(changeTenant(tenant.name));
     }
 
-    const { tenants } = manager;
-    if (!tenants || !tenants.items || tenants.isFetching) {
-        return <Loader active inverted inline size="small" />;
+    function handleTenantSearch(e: React.ChangeEvent<HTMLInputElement>, { value }: InputOnChangeData) {
+        e.stopPropagation();
+        setSearch(value);
     }
 
-    const filteredTenants = _(tenants.items)
+    const filteredTenants = _(tenants.items || [])
         .filter(tenant => _.includes(tenant.name, search))
         .sortBy('name')
         .value();
+
     const selectedTenant = tenants.selected || _.get(tenants, 'items[0].name');
 
     return (
-        <SideBarDropdownItem icon="user circle" label={selectedTenant || t('noTenants')} onClose={clearSearch}>
+        <SideBarDropdownItem
+            icon="user circle"
+            label={selectedTenant || t('noTenants')}
+            onClose={clearSearch}
+            onOpen={refreshTenants}
+        >
             <Dropdown.Header>{t('tenantsHeader')}</Dropdown.Header>
             <Dropdown.Menu scrolling>
-                {filteredTenants.map(tenant => (
-                    <Dropdown.Item
-                        key={tenant.name}
-                        text={tenant.name}
-                        selected={tenant.name === selectedTenant}
-                        active={tenant.name === selectedTenant}
-                        onClick={() => onTenantSelected(tenant)}
-                    />
-                ))}
+                {isLoadingTenants ? (
+                    <StyledLoaderWrapper>
+                        <Loading />
+                    </StyledLoaderWrapper>
```

Comment:
Optional:

Is this wrapper necessary? Wouldn't it be simpler to just have something like:
```suggestion
                    <Loading style={{ height: 80 }} />
```
I assume with a proper answer we need to wait for your investigation on React styling, am I correct?

Reply:
As for styling the component with the `styled-components`, it's required (without the wrapper, it's hard to determine the specifity of the css rule, without the `!important` flag)   

But I'm not blocked to use the coding methods which are related to our [UI Code Style](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/435388492/Cloudify+UI+Code+Style)

And because of that, and the fact the we are introducing here unnecessary wrapper (which you've pointed out 💪 ), I'll introduce the solution suggested by you 😉

Reply:
Done 🎉